pkg/sfu/downtrack.go

```diff
@@ -929,13 +925,40 @@ func (d *DownTrack) writeSimpleRTP(extPkt *buffer.ExtPacket) error {
 }
 
 func (d *DownTrack) writeSimulcastRTP(extPkt *buffer.ExtPacket, layer int32) error {
-	// Check if packet SSRC is different from before
-	// if true, the video source changed
+	tsl := d.TargetSpatialLayer()
 	csl := d.CurrentSpatialLayer()
+	if tsl == layer && csl != tsl {
+		if extPkt.KeyFrame {
+			d.currentSpatialLayer.set(layer)
+			csl = layer
+		} else {
+			d.lastPli.set(time.Now().UnixNano())
+			d.receiver.SendPLI(layer)
+		}
+	}
+
+	if tsl < csl {
+		//
+		// If target layer is lower, it is potentially due to bandwidth
```

Comment:
I'm not sure if this assumption is right. bandwidth limitation is definitely one of the scenarios where we are downgrading. What about for client requested downgrades? How do we distinguish server-driven downgrades from client-driven? maybe `if target < current && target < max` ?

Reply:
Good catch! I will make that change.

Reply:
@davidzhao Made the change in this commit - https://github.com/livekit/livekit-server/pull/203/commits/22988bec8d0c4a577d1b19d2f93140679197ee54. There are some edge cases. Added some more explanation and left a note about distinguishing server applied restriction vs client requested restriction. Will think about that case as it could be a tricky one, but should be a rare one (hopefully).